repose-aggregator/components/filters/header-normalization-filter/src/main/scala/org/openrepose/filters/headernormalization/HeaderNormalizationFilter.scala

```diff
@@ -71,22 +73,42 @@ class HeaderNormalizationFilter @Inject()(configurationService: ConfigurationSer
 
     val wrappedRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
 
-    config find { target =>
+    configRequest find { target =>
       // find the first "target" config element that matches this request (if any)
       target.url.matcher(wrappedRequest.getRequestURI).matches &&
```

Comment:
You're right that the `Pattern` is not cached, but going from `String` to `Pattern` should be pretty cheap.

If you wanted, you could use the `RegexString` class directly. To provide maximum flexibility, the `RegexStringOperators` file allows you to either import the implicit conversion to a regex string, mix-in the implicit conversion to a regex string, or instantiate a regex string. So you could build your regex strings on configuration updated, but...